evm-ext/src/solidity_abi.rs

```diff
@@ -69,11 +77,23 @@ pub fn abi_encode_params(
     let prefix = args.pop_back().unwrap().value_as::<Vec<u8>>()?;
     let ty_arg = safely_pop_type_arg!(ty_args);
 
-    // TODO: need to figure out how much gas to charge for these operations.
+    // Charge for the lookup of the type (twice because we need to get annotated and not).
```

Comment:
I'm sure this is correct, but I don't fully understand the explanation here. Is it twice for annotated and unannotated?

Reply:
Yeah we need to look up both layouts so we need to charge once for each. The users must pay for all resource expenditures. If we can find a more efficient implementation of this in the future then we could lower its gas cost.

Reply:
I think we don't have an issue for investigating this annotated layout not being supported, right? And I feel the aptos version bump didn't change things.

Reply:
There is no issue for it. Honestly I don't think it's a problem; the gas costs shouldn't be too bad even with the double type info lookup. Nothing about this has changed with the recent Aptos version bump, correct.